### PR TITLE
Document steps for cutting a new release

### PR DESCRIPTION
**Pull request by jeffwidman**
Document the steps to cut/tag a new release of this action.

### CODE REVIEW

File: README.md
Comment (deivid-rodriguez) [inline README.md:202]: I'm curious about why we need to do all the above. In my opinion, a tag should be fixed and not move. Forcing a tag to behave like a branch feels confusing. I understand that this is actually standard practice for handling many GitHub actions releases, but I'm curious about why. I have a feeling this practice may actually be widespread because of issues with Dependabot?

To clarify how I would expect Dependabot to work, I think someone using `dependabot/fetch-metadata@v1` in their workflow files should always get the latest v1.x.x release (tag) of the action, even if there's no tag named "v1". If this is how dependabot works, I think the better solution could be to actually delete the v1 and not run any of the four commands above when releasing.

Comment (landongrindheim) [reply README.md:202]: > In my opinion, a tag should be fixed and not move.

I understand this concern, but in this case we're conforming to the ecosystem's convention. GitHub has docs about [release management for GitHub Actions](https://docs.github.com/en/actions/creating-actions/about-custom-actions#good-practices-for-release-management), to which these instructions align.

To add some more context, here are the tags for one of the more popular Actions (`actions/checkout`):
image

Comment (deivid-rodriguez) [reply README.md:202]: Hi @landongrindheim 👋!

Thanks for pointing me to the official documentation, I knew this was standard practice, but I didn't know where this standard practice came from. It still doesn't make sense to me though.

Other actions that don't do this get the same end user behavior as far as I know, for example, https://github.com/ruby/setup-ruby. Users specifying `uses: ruby/setup-ruby@v1` also get the latest v1 release of the action, even if the action does not have a tag named "v1" always pointing to the latest v1 release. So I don't get the motivation for recommending this way of managing releases, in particular,

> Using this approach, your users can easily distinguish between major and minor versions

Am I missing something?

Comment (landongrindheim) [reply README.md:202]: @deivid-rodriguez I'm 100% certain you've thought more deeply about versioning than I have, so I don't be able to answer any questions re: whether this is a good approach 😅 I only know of this being a practice.

---

I looked at `ruby/setup-ruby` to see what was going on and saw that `v1` is a branch:

image

They don't have a `v1` tag: 

image

I don't know if this is a recommended practice, but I kinda like that their tags are immutable at least 🙂

Comment (deivid-rodriguez) [reply README.md:202]: Aaaaaah I missed that!

So, if `v1` was not a branch nor a tag, then `uses: ruby/setup-ruby@v1` would not work at all and raise an error, right? If that's the case, then that perfectly explains why we need to do this 😃!

It's true that `ruby/setup-ruby` tags are immutable, but it's also true that they are naming a branch using a name you usually expect to refer to a tag. So also confusing, although I like it better than the recommended approach of having moving tags.

I actually expected Github Actions to handle this "properly": if `v1` is not a tag nor a branch, but looks like a version, then automatically find the closest (semver wise) existing tag and use that.

Maybe I'll request this feature to Github Actions but I think my question is solved now, thanks @landongrindheim!

Comment (jeffwidman) [reply README.md:202]: I'm going to temp unresolve this as a reminder to myself to read up on this... and possibly raise the idea of converting `v1` to be a branch rather than a tag. I prefer immutable tags. I copied these instructions from one of our internal docs and didn't actually read them so didn't even realize this was being done this way/recommended til @deivid-rodriguez pointed out it.

Comment (deivid-rodriguez) [reply README.md:202]: Just documenting what I found when I was going to request this feature.

* Funny enough, the maintainer of `ruby/setup-ruby` already requested to change the official recommendation from moving tags to moving branches: https://github.com/actions/toolkit/pull/631. But [a similar request](https://github.com/actions/toolkit/issues/214) was already closed, so it doesn't seem like it will be changed.

* The, in my opinion, best solution since it does not require moving tags nor branches, is already requested at https://github.com/orgs/community/discussions/9847. I thumbed up and subscribed to see if there's some movement there.

Comment (jeffwidman) [reply README.md:202]: After reading through the links, I dislike both tags and branches... 😀 

I think right now we should stick with what the ecosystem standard is, which is _generally_ moving a `vX` tag... If that later changes, we can easily update this here as well.

Comment (deivid-rodriguez) [reply README.md:202]: Absolutely, my latest comments were not meant at blocking this PR!

**Review by deivid-rodriguez — APPROVED** [review]

File: README.md
Comment (jeffwidman) [reply README.md:202]: Cool. And FWIW, thanks for posting those links, I also subscribed, and the discussion was educational, I hadn't thought about the downsides tradeoffs of using a branch before this.

**Pull request merged**
